### PR TITLE
Add collection of cross package test coverage

### PR DESCRIPTION
**Pull request by chudilka1**
* Updated `go test` command to provide cumulative cross-package test coverage metrics (for GitHub actions)
* Excluded `mocks` packages from SonarQube analysis

### CODE REVIEW

File: tools/bin/go_core_tests [hunk 1]
```diff
@@ -8,7 +8,7 @@ OUTPUT_FILE="./output.txt"
 echo "Failed tests and panics: ---------------------"
 echo ""
 GO_LDFLAGS=$(bash tools/bin/ldflags)
-go test -ldflags "$GO_LDFLAGS" -tags integration -p 3 -coverprofile=coverage.txt -covermode=atomic $1 | tee $OUTPUT_FILE
+go test -ldflags "$GO_LDFLAGS" -tags integration -p 3 -covermode=count -coverpkg=./... -coverprofile=coverage.txt  $1 | tee $OUTPUT_FILE
```
Comment (jmank88) [inline tools/bin/go_core_tests:11]: Oh awesome, does `coverpkg` do exactly what we need?

Why did you change the mode from `atomic` to `count`? Is there a significant speedup?

**Review by jmank88 — DISMISSED** [review]

File: sonar-project.properties [hunk 2]
```diff
@@ -5,7 +5,7 @@ sonar.python.version=3.8
 # Full exclusions from the static analysis
 sonar.exclusions=**/node_modules/**/*, **/mocks/**/*, **/testdata/**/*, **/contracts/typechain/**/*, **/contracts/artifacts/**/*, **/contracts/cache/**/*, **/contracts/scripts/**/*, **/generated/**/*, **/fixtures/**/*, **/docs/**/*, **/tools/**/*, **/*.pb.go, **/*report.xml, **/*.config.ts, **/*.txt, **/*.abi, **/*.bin, **/*_codecgen.go
 # Coverage exclusions
-sonar.coverage.exclusions=**/*.test.ts, **/*_test.go, **/contracts/test/**/*, **/contracts/**/tests/**/*, **/core/**/testutils/**/*, **/core/**/cltest/**/*, **/integration-tests/**/* 
+sonar.coverage.exclusions=**/*.test.ts, **/*_test.go, **/contracts/test/**/*, **/contracts/**/tests/**/*, **/core/**/testutils/**/*, **/core/**/mocks/**/*, **/core/**/cltest/**/*, **/integration-tests/**/*
```
Comment (jmank88) [inline sonar-project.properties:8]:
```suggestion
sonar.coverage.exclusions=**/*.test.ts, **/*_test.go, **/contracts/test/**/*, **/contracts/**/tests/**/*, **/core/**/testutils/**/*, **/core/**/mocks/**/*, **/core/**/cltest/**/*, **/integration-tests/**/*, **/generated/**/*, **/core/scripts** 
```

File: tools/bin/go_core_tests (hunk 1 above)
Comment (chudilka1) [reply tools/bin/go_core_tests]: Thanks, I reverted to `atomic` as we have multi-threaded tests, and that one, though is slower, provides more accurate metrics.

**Review by jmank88 — DISMISSED** [review]

**Review by chainchad — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]
